presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -249,45 +251,45 @@ public QueryInfo getQueryInfo(StageInfo rootStage)
         long outputDataSize = 0;
         long outputPositions = 0;
 
-        boolean fullyBlocked = rootStage != null;
+        boolean fullyBlocked = rootStage.isPresent();
         Set<BlockedReason> blockedReasons = new HashSet<>();
 
-        if (rootStage != null) {
```

Comment:
getAllStages is public. It is used in other places as well. It's convenient for getAllStages() to take Optional, isn't it?